src/main/java/LocalKeyClientEncryption.java

```diff
@@ -0,0 +1,175 @@
+     */
+    public void decryptReupload(String storageAccount, String sharedKeyCred, String containerName, String blobName,
+                                String blobSuffix, AsyncKeyEncryptionKey key, CustomerProvidedKey serverKey){
+        String storageAccountUrl = "https://" + storageAccount + ".blob.core.windows.net";
+
+        // Creating blob client to download blob
+        BlobClient blobClient = new BlobClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .containerName(containerName)
+                .blobName(blobName + blobSuffix)
+                .buildClient();
+
+        // Set blob encryption client
+        if (key == null) {
+            key = this.key;
+        }
+        this.setLocalKeyClient(key, blobClient);
+
+        // Downloading encrypted blob, blob is decrypted upon download
+        String fileName = blobName + "Decrypted" + blobSuffix;
+        this.encryptedBlobClient.downloadToFile(".\\src\\main\\java\\" + fileName);
+
+        // Creating blob client for decryption and reuploading
+        BlobClientBuilder blobClientBuilder = new BlobClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .containerName(containerName)
+                .blobName(fileName);
+
+        // Checking to see if local key is provided to use customer-managed key for server encryption
+        if (serverKey != null){
+            String CPKfileName = blobName + "DecryptedCPK" + blobSuffix;
+            blobClientBuilder = blobClientBuilder
+                    .customerProvidedKey(serverKey)
+                    .blobName(CPKfileName);
+        }
+        BlobClient blobClientDecrypted = blobClientBuilder.buildClient();
+
+        // Uploading file to server
+        blobClientDecrypted.uploadFromFile(".\\src\\main\\java\\" + fileName);
+
+        // Cleaning up by deleting local save of encrypted blob
+        File localFile = new File(".\\src\\main\\java\\" + fileName);
+        localFile.delete();
+
+    }
+
+    /**
+     * Requires setting and naming the following environmental variables (must be in system environmental variables):
+     * sasToken (for storage account)
+     * storageAccount
+     * This sample will show client-side decryption using a local key, and then upload two files with server-side
+     * encryption. One will be encrypted with Microsoft-managed keys, and the other is by customer-provided keys.
+     */
+    public static void main(String[] args){
+        String sharedKeyCred = System.getenv("sharedKeyCred");
+        String storageAccount = System.getenv("storageAccount");
+
+        // Setting names of container and blob so that they match ones from setup. Can be changed in case where
+        // set up is not used
+
+        String containerName = "containername";
```

Comment:
Instead of redeclaring the container/blob, it may be better to pass them into setup as method params. If someone wants to edit the program, they won't have to edit it in both places that way